signal_scanner_bot/env.py

```diff
@@ -17,22 +17,42 @@ class _State:
     STOP_REQUESTED = False
 
 
-def _env(key: str, fail: bool = True, default: Any = None) -> Optional[str]:
+def _env(
+    key: str,
+    convert: Callable[[str], Any] = None,
+    fail: bool = True,
+    default: Any = None,
+) -> Optional[Any]:
     value = os.environ.get(key)
     if value is None:
         if fail and default is None:
             raise KeyError(f"Key '{key}' is not present in environment!")
         return default
+    if convert is not None:
+        value = convert(value)
     _VARS.append((key, value))
     return value
 
 
+def _cast_to_bool(to_cast: str) -> bool:
+    return to_cast.lower() == "true"
+
+
+def _cast_to_list(to_cast: str) -> List[str]:
+    return to_cast.split(",")
+
+
+def _cast_to_set(to_cast: str) -> Set[str]:
+    return set(to_cast.split(","))
+
+
 def log_vars() -> None:
     for key, value in _VARS:
         log.debug(f"{key}={value}")
 
 
-DEBUG = _env("DEBUG", default=False)
+TESTING = _env("TESTING", convert=_cast_to_bool, default=False)
+DEBUG = TESTING or _env("DEBUG", default=False)
```

Comment:
Should this have the cast to bool conversion as well?

Reply:
We sure should thanks for the catch.